### PR TITLE
Compressed MapGrid2D

### PR DESCRIPTION
**Pull request by randaz81**
#### This pr needs to be rebased as soon as #2448 is merged.

Relevant changes:
### yarp::dev
* added internal lossless compression for datatype `yarp::dev::Nav2D::MapGrid2D` using Zlib: the methods `yarp::os::Portable::write()`
  and `yarp::os::Portable::read()` handle the compression before sending data to the network.
* the method `MapGrid2D::enable_map_compression_over_network()` enables/disables the data compression over the network (default true).
* if Zlib library is not available, compression will be set to false.

### devices
* added rpc option `enable_maps_compression <0/1>` to `yarp::dev::Map2DServer`

### CODE REVIEW

File: src/devices/map2DServer/Map2DServer.cpp [hunk 1]
```diff
@@ -714,6 +714,12 @@ void Map2DServer::parse_string_command(yarp::os::Bottle& in, yarp::os::Bottle& o
         m_maps_storage.clear();
         out.addString("all maps cleared");
     }
+    else if (in.get(0).asString() == "enable_maps_compression")
+    {
+        for (auto it=m_maps_storage.begin(); it!= m_maps_storage.end(); it++)
+            {it->second.enable_map_compression_over_network(in.get(1).asBool());}
+        out.addString("compression mode of all maps set to:"+ in.get(1).asString());
```
Comment (drdanz) [inline src/devices/map2DServer/Map2DServer.cpp:721]: Please add a warning when compression is not enabled

**Review by drdanz — APPROVED** [review]
LGTM, just a comment

File: src/devices/map2DServer/Map2DServer.cpp (hunk 1 above)
Comment (randaz81) [reply src/devices/map2DServer/Map2DServer.cpp]: warning added

**Pull request merged**